qubes/tests/vm/dispvm.py

```diff
@@ -116,10 +116,16 @@ def test_002_template_change(self):
             dispvm = self.app.add_new_vm(qubes.vm.dispvm.DispVM,
                 name='test-dispvm', template=self.appvm)
 
-            with self.assertRaises(qubes.exc.QubesValueError):
-                dispvm.template = self.appvm
+            dispvm.template = self.appvm
+            dispvm.start()
+            if not self.app.vmm.offline_mode:
+                assert not dispvm.is_halted()
+                with self.assertRaises(qubes.exc.QubesVMNotHaltedError):
+                    dispvm.template = self.appvm
             with self.assertRaises(qubes.exc.QubesValueError):
                 dispvm.template = qubes.property.DEFAULT
+            dispvm.kill()
+            dispvm.template = self.appvm
```

Comment:
Since the whole thing was blocked because it caused volumes not being updated, can you add a test that verifies it? I think we need two cases:
1. 2x TemplateVM + 2x AppVM + DispVM, then change template of the DispVM and verify if its root and private volumes were updated.
2. 2x TemplateVM + AppVM + DispVM, then change template of the AppVM and verify if DispVM's root volume was updated.

There are already similar tests for AppVM, you can copy from there.